### PR TITLE
log refactor: relay

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
update log use pingcap/log

### What is changed and how it works?
update log

Pr recommended review order
https://github.com/pingcap/dm/pull/162
https://github.com/pingcap/dm/pull/170
https://github.com/pingcap/dm/pull/172
https://github.com/pingcap/dm/pull/173
https://github.com/pingcap/dm/pull/175
https://github.com/pingcap/dm/pull/176

### CODE REVIEW

File: relay/reader/reader.go [hunk 1]
```diff
@@ -139,12 +147,12 @@ func (r *reader) GetEvent(ctx context.Context) (*replication.BinlogEvent, error)
 
 func (r *reader) setUpReaderByGTID() error {
 	gs := r.cfg.GTIDs
-	log.Infof("[relay] start sync for master %s from GTID set %s", r.cfg.MasterID, gs)
+	logger.Info("start sync", zap.String("master", r.cfg.MasterID), zap.String("from GTID set", gs))
```
Comment (amyangfei) [inline relay/reader/reader.go]:
```suggestion
	logger.Info("start sync", zap.String("master", r.cfg.MasterID), zap.Stringer("from GTID set", gs))
```

File: relay/reader/reader.go [hunk 2]
```diff
@@ -139,12 +147,12 @@ func (r *reader) GetEvent(ctx context.Context) (*replication.BinlogEvent, error)
...
 	return r.in.StartSyncByGTID(gs)
 }
 
 func (r *reader) setUpReaderByPos() error {
 	pos := r.cfg.Pos
-	log.Infof("[relay] start sync for master %s from position %s", r.cfg.MasterID, pos)
+	logger.Info("start sync", zap.String("master", r.cfg.MasterID), zap.String("from position", pos))
```
Comment (amyangfei) [inline relay/reader/reader.go]:
```suggestion
	logger.Info("start sync", zap.String("master", r.cfg.MasterID), zap.Stringer("from position", pos))
```

File: relay/relay.go [hunk 3]
```diff
@@ -862,19 +870,19 @@ func (r *Relay) retrySyncGTIDs() error {
...
 	_, oldGTIDSet := r.meta.GTID()
-	log.Infof("[relay] start retry sync with old GTID %s", oldGTIDSet.String())
+	logger.Info("start retry sync with old GTID", zap.Stringer("GTID", oldGTIDSet))
 
 	_, newGTIDSet, err := utils.GetMasterStatus(r.db, r.cfg.Flavor)
 	if err != nil {
 		return errors.Annotatef(err, "get master status")
 	}
-	log.Infof("[relay] new master GTID set %v", newGTIDSet)
+	logger.Info("", zap.Reflect("new master GTID set", newGTIDSet))
```
Comment (amyangfei) [inline relay/relay.go]:
```suggestion
	logger.Info("", zap.Stringer("new master GTID set", newGTIDSet))
```

File: relay/relay.go [hunk 4]
```diff
@@ -313,15 +321,15 @@ func (r *Relay) process(parentCtx context.Context) error {
 			if err != nil {
 				return errors.Annotatef(err, "start to fill gap in relay log file from %v", gapSyncStartPos)
 			}
-			log.Infof("[relay] start to fill gap in relay log file from %v", gapSyncStartPos)
+			logger.Info("start to fill gap in relay log file", zap.Reflect("from", gapSyncStartPos))
```
Comment (amyangfei) [inline relay/relay.go]:
```suggestion
			logger.Info("start to fill gap in relay log file", zap.Stringer("from", gapSyncStartPos))
```

File: relay/relay.go [hunk 5]
```diff
@@ -355,24 +363,24 @@ func (r *Relay) process(parentCtx context.Context) error {
...
+							_, err2 = r.fd.Seek(int64(pos), io.SeekStart)                // seek to a valid pos
 							if err2 != nil {
 								return errors.Annotatef(err, "seek %s to pos %d", r.fd.Name(), pos)
 							}
 							err2 = r.fd.Truncate(int64(pos)) // truncate the incomplete event part
 							if err2 != nil {
 								return errors.Annotatef(err, "truncate %s to size %d", r.fd.Name(), pos)
 							}
-							log.Infof("[relay] truncate %s to size %d", r.fd.Name(), pos)
+							logger.Info("truncate file", zap.String("file", r.fd.Name()), zap.Reflect("to size", pos))
```
Comment (amyangfei) [inline relay/relay.go]:
```suggestion
							logger.Info("truncate file", zap.String("file", r.fd.Name()), zap.Uint32("to size", pos))
```

File: relay/relay.go [hunk 6]
```diff
@@ -355,24 +363,24 @@ func (r *Relay) process(parentCtx context.Context) error {
...
 							gapSyncStartPos = &mysql.Position{
 								Name: lastPos.Name,
 								Pos:  pos,
 							}
 							gapSyncEndPos = &gapSyncEndPos2 // reset the endPos
-							log.Infof("[relay] adjust gap streamer's start pos to %d", pos)
+							logger.Info("adjust gap streamer's start pos", zap.Reflect("pos", pos))
```
Comment (amyangfei) [inline relay/relay.go]:
```suggestion
							logger.Info("adjust gap streamer's start pos", zap.Uint32("pos", pos))
```

**Comment by GregoryIan** [comment]
@WangXiangUSTC I also modify it in #172, I think I should close it 😄

**Pull request closed without merging**